source/common/event/win32/signal_impl.cc

```diff
@@ -0,0 +1,40 @@
+#include "common/api/os_sys_calls_impl.h"
+#include "common/event/dispatcher_impl.h"
+#include "common/event/signal_impl.h"
+
+#include "event2/event.h"
+
+namespace Envoy {
+namespace Event {
+
+SignalEventImpl::SignalEventImpl(DispatcherImpl& dispatcher, signal_t signal_num, SignalCb cb)
+    : cb_(cb) {
+  auto handler_it = eventBridgeHandlersSingleton::get().find(signal_num);
+  if (handler_it != eventBridgeHandlersSingleton::get().end()) {
+    return;
+  }
+
+  os_fd_t socks[2];
+  Api::SysCallIntResult result =
+      Api::OsSysCallsSingleton::get().socketpair(AF_INET, SOCK_STREAM, IPPROTO_TCP, socks);
+  ASSERT(result.rc_ == 0);
+
+  read_handle_ = std::make_unique<Network::IoSocketHandleImpl>(socks[0], false, AF_INET);
+  result = read_handle_->setBlocking(false);
+  ASSERT(result.rc_ == 0);
+  auto write_handle = std::make_shared<Network::IoSocketHandleImpl>(socks[1], false, AF_INET);
+  result = write_handle->setBlocking(false);
+  ASSERT(result.rc_ == 0);
```

Comment:
possible future refactor: I wished that a socket pair wakeup mechanism existed as an primitive under source/common/event .  There's several scenarios where it could prove useful, including replacing the wakeup mechanism currently used for Dispatcher::post callbacks.

Reply:
I assume the wake up mechanism is: `post_cb_->scheduleCallbackCurrentIteration();`. Makes sense, I would like to get #10871 first so we can have `AF_UNIX` socketpairs on windows and then I can add that.

I will open an issue for that so that we don't forget it in the meantime

Reply:
It is.  Schedulable callbacks are implemented on top of libevent timers which require taking a lock on the event loop in order to trigger activation.

Wakeup via a socket pair feels more elegant.

Reply:
see #14244